crates/relayer/src/worker/channel.rs

```diff
@@ -49,19 +51,63 @@ pub fn spawn_channel_worker<ChainA: ChainHandle, ChainB: ChainHandle>(
 
                         complete_handshake_on_new_block = false;
                         if let Some(event_with_height) = last_event {
-                            retry_with_index(
-                                channel_handshake_retry::default_strategy(max_block_times),
-                                |index| match RelayChannel::restore_from_event(
-                                    chains.a.clone(),
-                                    chains.b.clone(),
-                                    event_with_height.event.clone(),
-                                ) {
-                                    Ok(mut handshake_channel) => handshake_channel
-                                        .step_event(&event_with_height.event, index),
-                                    Err(_) => RetryResult::Retry(index),
-                                },
-                            )
-                            .map_err(|e| TaskError::Fatal(RunError::retry(e)))
+                            match event_with_height.event.event_type() {
+                                IbcEventType::UpgradeInitChannel
```

Comment:
`Cancel` and `Timeout` are not handled here, is that on purpose?

Reply:
Hermes doesn't have to react on `UpgradeCancel` as this is emitted when the upgrade is cancelled.
But we might require the `UpgradeTimeout` as Hermes should cancel the Upgrade on the other end

Reply:
Done in c74f9e57e7650a6d65d044bd155e76bea1a76668